### PR TITLE
`ipfs repo version` command

### LINKED ISSUE #2571: `ipfs repo version` command
It would be really useful to have this work:

```
> cat ~/.ipfs/version
3
> ipfs repo version
ipfs repo version fs-repo@3
> ipfs repo version -q
fs-repo@3
```
because that way tools and applications can easily check the version of repos without making nasty assumptions.

---

may also be relevant to include this in the `ipfs repo stat` output:

```
> ipfs repo stat
NumObjects 	 288
RepoSize 	 6
RepoPath 	 /Users/jbenet/.ipfs
Version 	 fs-repo@3
```

### PR DESCRIPTION
**Pull request by pfista**
Fixes #2571

### CODE REVIEW

File: core/commands/repo.go [hunk 1]
```diff
@@ -105,6 +111,7 @@ set of stored objects and print repo statistics. It outputs to stdout:
 NumObjects      int Number of objects in the local repo.
 RepoPath        string The path to the repo being currently used.
 RepoSize        int Size in bytes that the repo is currently taking.
+Version         string The repo version
```
Comment (RichardLitt) [inline core/commands/repo.go]: Needs final `.`

**Comment by whyrusleeping** [comment]
@jbenet is this what you were after in #2571 ? (also note my comments there about `ipfs version --repo`)

**Comment by pfista** [comment]
Once I get confirmation if this is the right direction I'll update the tests as well.

**Comment by RichardLitt** [comment]
LGTM. @jbenet

**Comment by jbenet** [comment]
yeah LGTM, thanks

**Comment by Kubuxu** [comment]
In this PR non building commits were merged but now it is too late.

**Comment by pfista** [comment]
Sorry, was expecting these to be squashed before merge!

**Comment by Kubuxu** [comment]
It isn't your fault in any case, we should have noticed.

**Comment by jbenet** [comment]
- I'm sorry, i didn't notice those. 
- We should not make it possible (or "so green") to merge if some commits have not passed tests...
- Also we should consider squashing them. (yes, rewriting history)

File: test/sharness/t0080-repo.sh [hunk 2]
```diff
@@ -244,6 +245,21 @@ test_expect_success "repo stats are updated correctly" '
...
+
+test_expect_success "repo version came out correct" '
+	egrep "^ipfs repo version fs-repo@[0-9]" repo-version >/dev/null
+'
+
+test_expect_success "'ipfs repo version -q' succeeds" '
+  ipfs repo version -q > repo-version-q
+'
+test_expect_success "repo version came out correct" '
+	egrep "^fs-repo@[0-9]" repo-version-q >/dev/null
```
Comment (jbenet) [inline test/sharness/t0080-repo.sh]: this may need to be `fs-repo@[0-9]+`

File: test/sharness/t0080-repo.sh [hunk 3]
```diff
@@ -244,6 +245,21 @@ test_expect_success "repo stats are updated correctly" '
   test $(get_field_num "RepoSize" repo-stats-2) -ge $(get_field_num "RepoSize" repo-stats)
 '
 
+test_expect_success "'ipfs repo version' succeeds" '
+  ipfs repo version > repo-version
+'
+
+test_expect_success "repo version came out correct" '
+	egrep "^ipfs repo version fs-repo@[0-9]" repo-version >/dev/null
```
Comment (jbenet) [inline test/sharness/t0080-repo.sh]: this may need to be `fs-repo@[0-9]+`

**Comment by Kubuxu** [comment]
I will follow up.

**Pull request merged**